src/custom/xdai/declarations.d.ts

```diff
@@ -0,0 +1,212 @@
+import {
+  Currency,
+  CurrencyAmount,
+  BigintIsh,
+  Price,
+  TradeType,
+  Percent,
+  BestTradeOptions,
+  TradeOptions,
+  TradeOptionsDeadline,
+  SwapParameters
+} from '@uniswap/sdk'
+import { ChainId } from './'
+
+declare module '@uniswap/sdk' {
+  export class Token extends Currency {
+    readonly chainId: ChainId
```

Comment:
Replaced `ChainId` from `@uniswap/sdk` with our own, which isn't immediately compatible with their types, so have to override + override types that use overridden type :confounded:

Reply:
All the types are copypasted